apps/taxes/src/modules/avatax/tax-code/avatax-tax-code-matcher.ts

```diff
@@ -0,0 +1,26 @@
+import { TaxBaseLineFragment } from "../../../../generated/graphql";
+import { AvataxTaxCodeMatches } from "./avatax-tax-code-match-repository";
+
+export class AvataxTaxCodeMatcher {
+  private mapTaxClassWithTaxMatch(taxClassId: string, matches: AvataxTaxCodeMatches) {
+    return matches.find((m) => m.data.saleorTaxClassId === taxClassId);
+  }
+
+  private getTaxClassId(line: TaxBaseLineFragment): string | undefined {
+    if (line.sourceLine.__typename === "CheckoutLine") {
+      return line.sourceLine.checkoutProductVariant.product.taxClass?.id;
+    }
+
+    if (line.sourceLine.__typename === "OrderLine") {
+      return line.sourceLine.orderProductVariant?.product.taxClass?.id;
+    }
+  }
+
+  match(line: TaxBaseLineFragment, matches: AvataxTaxCodeMatches) {
+    const taxClassId = this.getTaxClassId(line);
+
+    return taxClassId
+      ? this.mapTaxClassWithTaxMatch(taxClassId, matches)?.data.avataxTaxCode ?? ""
+      : "";
```

Comment:
are you sure this should fallback to empty string? Avatax needs empty string in case of missing ID/code?

Reply:
yes

Reply:
So lets:
1. Add a comment describing this behavior
2. Ensure this code lands directly into avatar request (don't push application code with empty strings - better use nulls etc, and map to empty string when request is created)

Reply:
> Ensure this code lands directly into avatar request (don't push application code with empty strings - better use nulls etc, and map to empty string when request is created)

Not sure if I understand. This code lands directly in the payload of Avatax request

Reply:
So thats fine.

What I meant is that we can have different model in app's codebase and we can map it later in the request. This allows us to write maintainable data models and stringify them in the intersection points (req, res)